src/Week10/AParser.hs

```diff
@@ -0,0 +1,75 @@
+module Week10.AParser where
+
+import Control.Applicative
+import Control.Monad
+import Data.Char
+
+------------------------------------------------------------
+-- Provided code
+------------------------------------------------------------
+newtype Parser a = Parser { runParser :: String -> Maybe (a, String) }
+
+satisfy :: (Char -> Bool) -> Parser Char
+satisfy p = Parser f
+  where
+    f [] = Nothing
+    f (x:xs)
+        | p x       = Just (x, xs)
+        | otherwise = Nothing
+
+char :: Char -> Parser Char
+char c = satisfy (== c)
+
+posInt :: Parser Integer
+posInt = Parser f
+  where
+    f xs
+      | null ns   = Nothing
+      | otherwise = Just (read ns, rest)
+      where (ns, rest) = span isDigit xs
+
+------------------------------------------------------------
+-- Homework code begins here
+------------------------------------------------------------
+
+---------- ex 1 ----------
+first :: (a -> b) -> (a,c) -> (b,c)
+first f (a,c) = (f a, c)
```

Comment:
In real life you don't need to implement such `first` function manually because it comes from `Bifunctor` module:

* https://hackage.haskell.org/package/base-4.11.1.0/docs/Data-Bifunctor.html#v:first